object_manager.go

```diff
@@ -317,20 +336,35 @@ func (objMgr *ObjectManager) GetFixedAddress(netview string, cidr string, ipAddr
 	return &res[0], nil
 }
 
-func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, macAddress string, name string, vmID string) (*FixedAddress, error) {
+func (objMgr *ObjectManager) GetFixedAddressFromref(ref string) (*FixedAddress, error) {
+        fixedAddr := NewFixedAddress(FixedAddress{})
+        err := objMgr.connector.GetObject(fixedAddr, ref, &fixedAddr)
+        return fixedAddr, err
+}
+
+func (objMgr *ObjectManager) DeleteFixedAddress(ref string) (string, error) {
+        return objMgr.connector.DeleteObject(ref)
+}
+
+func (objMgr *ObjectManager) UpdateFixedAddress(fixedAddrRef string, matchclient string, macAddress string, vmID string, vmName string) (*FixedAddress, error) {
 
 	updateFixedAddr := NewFixedAddress(FixedAddress{Ref: fixedAddrRef})
 
 	if len(macAddress) != 0 {
 		updateFixedAddr.Mac = macAddress
 	}
 
-	if vmID != "" {
+	if vmID != "" &&vmName != ""{
```

Comment:
This could be `||`, is it not ? or are you expecting both not empty ?

Reply:
actually i am expecting both to be not empty , because even if one of them is passed empty , it raises wapi error.
Tested when i made it ||
case 1 : passed vmID but no name
 "Error": "AdmConDataError: None (IBDataConflictError: IB.Data.Conflict:Missing value for extensible attribute 'VM Name'.)",

case 2 :passed vmName but no ID
 "Error": "AdmConDataError: None (IBDataConflictError: IB.Data.Conflict:Missing value for extensible attribute 'VM ID'.)",

Reply:
ok then it should be like this
```suggestion
	if vmID != "" {
	   ea["VM ID"] = vmID
	}
	if vmName != ""{
	   ea["VM Name"] = vmName
	}
```